addons/stock_account/models/stock.py

```diff
@@ -150,20 +177,29 @@ def replay(self):
 
     @api.multi
     def action_done(self):
+        """
+            For incoming moves, it will put the value corresponding with that move on the stock move, 
+            except for standard cost price, where it will put the standard price.  
+            For FIFO, it will also check if there was negative stock and compensate the corresponding out moves.   
+            For outgoing moves, average price, will apply the average price formula, FIFO will search the 
+            corresponding incoming moves where it can reduce the remaining_qty while standard takes the cost from the product
+        """
         qty_available = {}
         for move in self:
-            #Should write move.price_unit here maybe, certainly on incoming
+            #Should write move.price_unit here maybe, certainly on incoming moves
             if move.product_id.cost_method == 'average':
-                qty_available[move.product_id.id] = move.product_id.qty_available
+                qty_available[move.product_id.id] = move.product_id.with_context(internal=True).qty_available
         res = super(StockMove, self).action_done()
         for move in res:
-            if move.location_id.usage not in ('internal', 'transit') and move.location_dest_id.usage in ('internal', 'transit'):
+            if move._is_in_move():
                 if move.product_id.cost_method in ['fifo', 'average']:
                     if not move.price_unit:
                         move.price_unit = move._get_price_unit()
-                    move.value = move.price_unit * move.product_qty
-                    move.cumulated_value = move.product_id._get_latest_cumulated_value(not_move=move) + move.value
-                    move.remaining_qty = move.product_qty
+                    move_value = move.price_unit * move.product_qty
+                    move.write({'value': move_value, 
+                                'cumulated_value': move.product_id._get_latest_cumulated_value(not_move=move) + move_value, 
+                                'remaining_qty': move.product_qty, 
+                                'last_done_qty': move.product_id.with_context(internal=True).qty_available,})
```

Comment:
acutally last_done_qty is not used in this commit, it'll be used by a further one that fix average edit in the past